.github/actions/hatch-setup/action.yml

```diff
@@ -61,36 +61,69 @@ runs:
           ${{ env.HATCH_DATA_DIR }}
         key: ${{ runner.os }}-hatch-${{ hashFiles('pyproject.toml') }}-${{ hashFiles('requirements.txt') }}
 
-    - name: Install Hatch Python
-      shell: bash
+    - name: Install Hatch Matrix Python Versions
+      id: setup-python
       if: steps.cache.outputs.cache-hit != 'true' && steps.test-matrix-present.outputs.matrix-present == 'true'
-      run: |
-        hatch python install ${{ inputs.min-python-version }} ${{ inputs.max-python-version }}
+      uses: actions/setup-python@v5.5.0
+      with:
+        python-version: |
+          ${{ inputs.min-python-version }}
+          ${{ inputs.max-python-version }}
+        cache: "pip"
+        cache-dependency-path: "pyproject.toml"
 
-    - name: Install Hatch Python
-      shell: bash
+    - name: Install Hatch Matrix Python Versions
```

Comment:
Same Step name, suggestion to make it unique if this is for the default case.

Reply:
Renamed steps